WooCommerce/Classes/ViewRelated/Products/Media/ProductImagesGalleryViewController.swift

```diff
@@ -0,0 +1,205 @@
+                                      preferredStyle: .alert)
+        alert.view.tintColor = .text
+        let cancel = UIAlertAction(title: NSLocalizedString(
+            "Cancel",
+            comment: "Dismiss button on the alert when the user taps to delete a Product image"
+        ), style: .cancel, handler: nil)
+
+        let delete = UIAlertAction(title: NSLocalizedString(
+            "Remove",
+            comment: "Confirm button on the alert when the user taps to delete a Product image"
+        ), style: .destructive) { [weak self] _ in
+            guard let self = self else {
+                return
+            }
+            if let index = self.currentImageIndex {
+                self.onDeletion(self.productImages[index])
+                self.productImages.remove(at: index)
+                self.collectionView.reloadData()
+            }
+
+            if self.productImages.count == 0 {
+                self.navigationController?.popViewController(animated: true)
+            }
+        }
+
+        alert.addAction(cancel)
+        alert.addAction(delete)
+
+        ServiceLocator.analytics.track(.productImageSettingsDeleteImageButtonTapped)
+
+        present(alert, animated: true, completion: nil)
+    }
+}
+
+// MARK: CollectionView DataSource
+//
+extension ProductImagesGalleryViewController: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return productImages.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let image = productImages[indexPath.row]
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: ProductImageCollectionViewCell.reuseIdentifier,
+                                                      for: indexPath)
+        configureRemoteImageCell(cell, productImage: image)
+        return cell
+    }
+
+}
+
+extension ProductImagesGalleryViewController: UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return collectionView.frame.size
+    }
+
+    // Scroll to the right index every time viewWillTransitionToSize is called.
+    // Then performBatchUpdates to adjust our layout.
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        scrollToSelectedIndex(index: currentImageIndex)
+        collectionView.reloadData()
+    }
+}
+
+// MARK: Cell configuration
+//
+private extension ProductImagesGalleryViewController {
+
+    func configureRemoteImageCell(_ cell: UICollectionViewCell, productImage: ProductImage) {
+        guard let cell = cell as? ProductImageCollectionViewCell else {
+            fatalError()
```

Comment:
Wouldn't it be nicer if we just don't configure the cell rather than crashing? 😅